examples/refrigerator-app/silabs/src/RefrigeratorManager.cpp

```diff
@@ -0,0 +1,207 @@
+    SetParentEndpointForEndpoint(kColdCabinetEndpointId, kRefEndpointId);
+    SetParentEndpointForEndpoint(kFreezeCabinetEndpointId, kRefEndpointId);
+
+    // set TagList
+    SetTagList(kColdCabinetEndpointId, Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(refrigeratorTagList));
+    SetTagList(kFreezeCabinetEndpointId, Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(freezerTagList));
+
+    app::Clusters::TemperatureControl::SetInstance(&sAppSupportedTemperatureLevelsDelegate);
+    return CHIP_NO_ERROR;
+}
+
+int8_t RefrigeratorManager::ConvertToPrintableTemp(int16_t temperature)
+{
+    constexpr uint8_t kRoundUpValue = 50;
+
+    // Round up the temperature as we won't print decimals on LCD
+    // Is it a negative temperature
+    if (temperature < 0)
+    {
+        temperature -= kRoundUpValue;
+    }
+    else
+    {
+        temperature += kRoundUpValue;
+    }
+
+    return static_cast<int8_t>(temperature / 100);
+}
+
+void RefrigeratorManager::RefAndTempCtrlAttributeChangeHandler(EndpointId endpointId, AttributeId attributeId, uint8_t * value,
+                                                               uint16_t size)
+{
+    switch (attributeId)
+    {
+    case RefAndTempAttr::CurrentMode::Id: {
+        int16_t currentMode = static_cast<int16_t>(*value);
+        mCurrentMode        = currentMode;
+    }
+    break;
+
+    case RefAndTempAttr::StartUpMode::Id: {
+        int16_t startUpMode = static_cast<int16_t>(*value);
+        mStartUpMode        = startUpMode;
+    }
+    break;
+
+    case RefAndTempAttr::OnMode::Id: {
+        int16_t onMode = static_cast<int16_t>(*value);
+        mOnMode        = onMode;
+    }
+    break;
+
+    default: {
+        ChipLogError(AppServer, "Unhandled Refrigerator and Temperature attribute %ld", attributeId);
+        return;
+    }
+    break;
+    }
+}
+
+void RefrigeratorManager::TempCtrlAttributeChangeHandler(EndpointId endpointId, AttributeId attributeId, uint8_t * value,
+                                                         uint16_t size)
+{
+    switch (attributeId)
+    {
+    case TempCtrlAttr::TemperatureSetpoint::Id: {
+        int16_t temperatureSetpoint = ConvertToPrintableTemp(static_cast<int16_t>(*value));
+        mTemperatureSetpoint        = temperatureSetpoint;
+        TempCtrlAttr::TemperatureSetpoint::Set(endpointId, temperatureSetpoint);
+    }
+    break;
+
+    case TempCtrlAttr::MinTemperature::Id: {
+        int16_t minTemperature = ConvertToPrintableTemp(static_cast<int16_t>(*value));
+        mMinTemperature        = minTemperature;
+        TempCtrlAttr::MinTemperature::Set(endpointId, minTemperature);
+    }
+    break;
+
+    case TempCtrlAttr::MaxTemperature::Id: {
+        int16_t maxTemperature = ConvertToPrintableTemp(static_cast<int16_t>(*value));
+        mMaxTemperature        = maxTemperature;
+        TempCtrlAttr::MaxTemperature::Set(endpointId, maxTemperature);
+    }
```

Comment:
This is for TN feature, but from the .Matter file, it's clear it's TL feature.

Reply:
Yeah updated the feature map value for wifi devices